### PR TITLE
ENH: pcds-5.8.3

### LINKED ISSUE #322: extra-install-steps do not run on apply_release
These should run prior to write-protecting the environment.

### LINKED ISSUE #323: Audit summary markdown table needs to strip whitespace characters from external descriptions
I saw this today, which fails to render properly from the markdown. Note the excessive whitespace and two trailing newline characters in the `black` vulnerability's description.

```
pip audit results:
|   name  | version | fix_version | new or seen |          id         |                                                                       desc                                                                       |                           are we ok                           |
|:-------:|:-------:|:-----------:|:-----------:|:-------------------:|:------------------------------------------------------------------------------------------------------------------------------------------------:|:-------------------------------------------------------------:|
|  black  |  24.2.0 |    24.3.0   |     New     |    PYSEC-2024-48    |                                                                         Versions of the package black before 24.3.0 are vulnerable to Regular Expression Denial of Service (ReDoS) via the lines_with_leading_tabs_expanded function in the strings.py file. An attacker could exploit this vulnerability by crafting a malicious input that causes a denial of service.

Exploiting this vulnerability is possible when running Black on untrusted input, or if you habitually put thousands of leading tab characters in your docstrings.                                                                          |                      Needs investigation                      |
|  ecdsa  |  0.18.0 |             |     Seen    | GHSA-wj6h-64fc-37mp |                                                  Minerva timing attack on P-256 in python-ecdsa                                                  | Not used in prod, will never be fixed, see tiled dependencies |
| ipython |  8.4.0  |     8.10    |     Seen    | GHSA-29gw-9793-fvw7 |                                            IPython vulnerable to command injection via set_term_title                                            |                          Windows-only                         |
| numexpr |  2.8.7  |             |     Seen    |    PYSEC-2023-163   | An issue in LanChain-ai Langchain v.0.0.245 allows a remote attacker to execute arbitrary code via the evaluate function in the numexpr library. |                  Only affects langchain users                 |

```

### LINKED ISSUE #324: Linux builds not fully compatible with rocky9
For example, the `perl` that gets installed here won't run on our rocky9 build host:
```
(pcds-5.8.2)zlentz@psbuild-rocky9-01:~$ perl -h
perl: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory
```

Need to find out if this is a specific error with specific packages or a more widespread issue

### PR DESCRIPTION
**Pull request by ZLLentz**
New env for April
I think we'll release/deploy this next week (week of 4/22) during the PAMM

closes #322 
closes #323
closes #324 

PCDS Package Updates
--------------------

|   Package    |  Old   |  New   |                        Release Notes                         |
|:------------:|:------:|:------:|:------------------------------------------------------------:|
| hutch-python | 1.18.5 | 1.20.0 | https://github.com/pcdshub/hutch-python/releases/tag/v1.19.0<br>https://github.com/pcdshub/hutch-python/releases/tag/v1.20.0 |
|  lightpath   | 1.0.3  | 1.0.4  |   https://github.com/pcdshub/lightpath/releases/tag/v1.0.4   |
|   pcdscalc   | 0.5.0  | 0.5.1  |   https://github.com/pcdshub/pcdscalc/releases/tag/v0.5.1    |
| pcdsdevices  | 8.3.0  | 8.4.0  |  https://github.com/pcdshub/pcdsdevices/releases/tag/v8.4.0  |
|    typhos    | 3.1.0  | 3.1.1  |    https://github.com/pcdshub/typhos/releases/tag/v3.1.1     |

SLAC Package Updates
--------------------

| Package |  Old   |  New   |                    Release Notes                     |
|:-------:|:------:|:------:|:----------------------------------------------------:|
|   pydm  | 1.22.0 | 1.22.1 | https://github.com/slaclab/pydm/releases/tag/v1.22.1 |

Python Community Core Package Updates
-------------------------------------

|  Package   |  Old  |  New  |
|:----------:|:-----:|:-----:|
| pre-commit | 3.6.2 | 3.7.0 |
|   pytest   | 8.0.2 | 8.1.1 |

Added the Following Packages
----------------------------

- flask-authnz
- pmpsui

Added the Following Dependencies
--------------------------------

- libxcrypt (required by perl)

### CODE REVIEW

**Comment by ZLLentz** [comment]
I found some errors in my update to the apply_release.sh script so I'm working through those
Otherwise this is looking pretty good

File: envs/pcds/env.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: pcds-5.8.2
+name: pcds-test
```
Comment (tangkong) [inline envs/pcds/env.yaml:1]:
```suggestion
name: pcds-5.8.3
```
?

**Review by tangkong — COMMENTED** [review]
lgtm on the whole

File: envs/pcds/env.yaml (hunk 1 above)
Comment (ZLLentz) [reply envs/pcds/env.yaml:1]: Good catch, I pilfered this yaml from the CI build and didn't change the name here

File: envs/pcds/env.yaml [hunk 2]
```diff
@@ -865,4 +866,4 @@ dependencies:
       - watchgod==0.8.2
       - websockets==11.0.3
       - xraylib==4.1.3
-prefix: /cds/home/z/zlentz/miniconda3/envs/pcds-5.8.2
+prefix: /home/runner/miniconda/envs/pcds-test
```
Comment (ZLLentz) [inline envs/pcds/env.yaml:869]:
```suggestion
prefix: /home/runner/miniconda/envs/pcds-5.8.3
```

**Comment by tangkong** [comment]
I'm actually not sure what the failures in `py310-next-full` etc are.  The tests all pass, and the security checks are similar to other passing tests.  Is it the pip-check failures?

On the plus side py310 and py311 are looking good!

**Comment by ZLLentz** [comment]
Yeah, I think those are just pip check errors- some weird conflict between conda and pypi installs that result in an inconsistent env at the end.

**Comment by ZLLentz** [comment]
I'm going to check what happened with the local tests I ran before leaving on Friday before going ahead with this

**Comment by ZLLentz** [comment]
Local tests all passed (barring race conditions in hutch-python and typhos)
One last local sec check for good luck and then sending it

**Pull request merged**